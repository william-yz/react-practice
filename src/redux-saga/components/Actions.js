import React from 'react'

import { Button } from 'antd'

export default ({ addTodo }) => {
  return (
    <Button type="primary" onClick={addTodo} >Add</Button>
  )
}
