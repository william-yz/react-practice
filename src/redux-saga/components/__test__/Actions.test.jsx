import React from 'react'
import renderer from 'react-test-renderer'

import Actions from '../Actions'

test('Actions case 1', () => {
  const addTodo = jest.fn()
  const redo = jest.fn()
  const someoneIsEditting = true
  const redoabled = true
  const component = renderer.create(<Actions
    addTodo={addTodo}
    redo={redo}
    someoneIsEditting={someoneIsEditting}
    redoabled={redoabled}
  />)
  const actions = component.toJSON()
  expect(actions).toMatchSnapshot()
})

test('Actions case 2', () => {
  const addTodo = jest.fn()
  const redo = jest.fn()
  const someoneIsEditting = true
  const redoabled = false
  const component = renderer.create(<Actions
    addTodo={addTodo}
    redo={redo}
    someoneIsEditting={someoneIsEditting}
    redoabled={redoabled}
  />)
  const actions = component.toJSON()
  expect(actions).toMatchSnapshot()
})

test('Actions case 3', () => {
  const addTodo = jest.fn()
  const redo = jest.fn()
  const someoneIsEditting = false
  const redoabled = true
  const component = renderer.create(<Actions
    addTodo={addTodo}
    redo={redo}
    someoneIsEditting={someoneIsEditting}
    redoabled={redoabled}
  />)
  const actions = component.toJSON()
  expect(actions).toMatchSnapshot()
})

test('Actions case 4', () => {
  const addTodo = jest.fn()
  const redo = jest.fn()
  const someoneIsEditting = false
  const redoabled = false
  const component = renderer.create(<Actions
    addTodo={addTodo}
    redo={redo}
    someoneIsEditting={someoneIsEditting}
    redoabled={redoabled}
  />)
  const actions = component.toJSON()
  expect(actions).toMatchSnapshot()
})
