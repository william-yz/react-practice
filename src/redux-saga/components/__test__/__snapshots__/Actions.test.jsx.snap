exports[`test Actions case 1 1`] = `
<span>
  <button
    className="ant-btn ant-btn-primary"
    disabled={true}
    onClick={[Function]}
    onMouseUp={[Function]}
    type="button">
    <span>
      Add
    </span>
  </button>
   
  <button
    className="ant-btn ant-btn-primary"
    disabled={true}
    onClick={[Function]}
    onMouseUp={[Function]}
    type="button">
    <span>
      Redo
    </span>
  </button>
</span>
`;

exports[`test Actions case 2 1`] = `
<span>
  <button
    className="ant-btn ant-btn-primary"
    disabled={true}
    onClick={[Function]}
    onMouseUp={[Function]}
    type="button">
    <span>
      Add
    </span>
  </button>
   
  <button
    className="ant-btn ant-btn-primary"
    disabled={true}
    onClick={[Function]}
    onMouseUp={[Function]}
    type="button">
    <span>
      Redo
    </span>
  </button>
</span>
`;

exports[`test Actions case 3 1`] = `
<span>
  <button
    className="ant-btn ant-btn-primary"
    disabled={false}
    onClick={[Function]}
    onMouseUp={[Function]}
    type="button">
    <span>
      Add
    </span>
  </button>
   
  <button
    className="ant-btn ant-btn-primary"
    disabled={false}
    onClick={[Function]}
    onMouseUp={[Function]}
    type="button">
    <span>
      Redo
    </span>
  </button>
</span>
`;

exports[`test Actions case 4 1`] = `
<span>
  <button
    className="ant-btn ant-btn-primary"
    disabled={false}
    onClick={[Function]}
    onMouseUp={[Function]}
    type="button">
    <span>
      Add
    </span>
  </button>
   
  <button
    className="ant-btn ant-btn-primary"
    disabled={true}
    onClick={[Function]}
    onMouseUp={[Function]}
    type="button">
    <span>
      Redo
    </span>
  </button>
</span>
`;
